### PR TITLE
Find a working combination of packages

### PR DESCRIPTION
**Pull request by milljm**
This is a work in progress in an attempt to find a working compatible set of packages to satisfy paraview and moose-dev (seems to come down to hdf5)

Refs #24661

### CODE REVIEW

**Comment by milljm** [comment]
I'll generate hashes for versioner when things appear to be working...

File: conda/mpich/meta.yaml [hunk 1]
```diff
@@ -67,6 +61,12 @@ requirements:
     - {{ moose_xorg_x11 }}      # [linux]
     - {{ moose_libglu }}        # [linux]
     - {{ moose_mesalib }}       # [linux]
+    - cmake
+    - make
+    - libtool
+    - autoconf
+    - automake 1.16.1
+    - m4
```
Comment (milljm) [inline conda/mpich/meta.yaml:69]: I just moved this around for visual appearance in the file.

File: conda/peacock/conda_build_config.yaml [hunk 2]
```diff
@@ -1,20 +1,5 @@
...
-moose_qt:
-  - qt-main 5.15.4
-
-moose_matplotlib:
-  - matplotlib 3.5.2
-
-moose_vtk:
-  - vtk 9.1.0
+moose_mpich:
+  - moose-mpich 4.1.1 build_0
```
Comment (milljm) [inline conda/peacock/conda_build_config.yaml:2]: Finally :)

File: conda/mpich/meta.yaml (hunk 1 above)
Comment (cticenhour) [reply conda/mpich/meta.yaml:69]: ✨ 👀

File: modules/contact/test/tests/mortar_cartesian_lms/tests [hunk 3]
```diff
@@ -10,6 +10,7 @@
     max_parallel = 4
     min_ad_size = '50'
     valgrind = 'heavy'
+    installation_type = in_tree
```
Comment (lindsayad) [inline modules/contact/test/tests/mortar_cartesian_lms/tests:13]: Why is this needed?

**Comment by lindsayad** [comment]
Is this PR in a draft state?

**Comment by milljm** [comment]
Its almost ready. Just this pesky Sandbox thing I've reported internal.

File: modules/contact/test/tests/mortar_cartesian_lms/tests (hunk 3 above)
Comment (milljm) [reply modules/contact/test/tests/mortar_cartesian_lms/tests:13]: This test does not seem to work when ran out of tree. That is, `./configure --prefix /some/path; make install`

File: conda/moose-dev/conda_build_config.yaml [hunk 4]
```diff
@@ -2,7 +2,10 @@ moose_libmesh:
   - moose-libmesh 2023.07.26 build_1
 
 moose_wasp:
-  - moose-wasp 2023.05.01
+  - moose-wasp 2023.07.17
```
Comment (cticenhour) [inline conda/moose-dev/conda_build_config.yaml:5]:
```suggestion
  - moose-wasp 2023.08.17
```

File: conda/moose-dev/conda_build_config.yaml [hunk 5]
```diff
@@ -2,7 +2,10 @@ moose_libmesh:
...
 
 moose_tools:
-  - moose-tools 2023.05.12
+  - moose-tools 2023.07.17
```
Comment (cticenhour) [inline conda/moose-dev/conda_build_config.yaml:8]:
```suggestion
  - moose-tools 2023.08.17
```

File: conda/moose-dev/conda_build_config.yaml [hunk 6]
```diff
@@ -2,7 +2,10 @@ moose_libmesh:
...
+
+moose_peacock:
+  - moose-peacock 2023.07.17
```
Comment (cticenhour) [inline conda/moose-dev/conda_build_config.yaml:11]:
```suggestion
  - moose-peacock 2023.08.17
```

File: conda/moose/conda_build_config.yaml [hunk 7]
```diff
@@ -1,11 +1,5 @@
-moose_libmesh:
- - moose-libmesh 2023.07.26 build_1
-
-moose_wasp:
-  - moose-wasp 2023.05.01
-
-moose_python:
- - python 3.9
+moose_dev:
+  - moose-dev 2023.08.07
```
Comment (cticenhour) [inline conda/moose/conda_build_config.yaml:2]:
```suggestion
  - moose-dev 2023.08.17
```

**Review by cticenhour — CHANGES_REQUESTED** [review]
There were several inconsistencies in package version definitions causing a failure on conda build for `moose-dev`, so I've put all I saw in suggestions below.

File: conda/template/conda_build_config.yaml [hunk 8]
```diff
@@ -1,11 +1,5 @@
-moose_libmesh:
- - moose-libmesh 2023.07.26 build_0
-
-moose_wasp:
- - moose-wasp 2023.05.01
-
-moose_python:
- - python 3.9
+moose_dev:
+  - moose-dev 2023.07.31
```
[the remark this answers is not in the document]
Comment (milljm) [reply conda/template/conda_build_config.yaml:2]: lol, woops.

👍

File: modules/contact/test/tests/mortar_cartesian_lms/tests (hunk 3 above)
Comment (cticenhour) [reply modules/contact/test/tests/mortar_cartesian_lms/tests:13]: @lindsayad any more comment here? This package update is getting close to being ready to merge - just turned on the full complement of `next` testing.

File: framework/include/utils/SymmetricRankTwoTensor.h [hunk 9]
```diff
@@ -144,7 +144,7 @@ class SymmetricRankTwoTensorTempl
 
 public:
   /// Copy assignment operator must be defined if used
-  SymmetricRankTwoTensorTempl(const SymmetricRankTwoTensorTempl<T> & a) = default;
+  // SymmetricRankTwoTensorTempl(const SymmetricRankTwoTensorTempl<T> & a) = default;
```
Comment (cticenhour) [inline framework/include/utils/SymmetricRankTwoTensor.h:147]: Did you mean to leave this commented out, rather than removing it?

File: modules/contact/test/tests/mortar_cartesian_lms/tests (hunk 3 above)
Comment (lindsayad) [reply modules/contact/test/tests/mortar_cartesian_lms/tests:13]: I just have no idea why this test should be different than any other. But it's definitely not important enough to me to prevent this moving forward

File: framework/include/utils/SymmetricRankTwoTensor.h (hunk 9 above)
Comment (milljm) [reply framework/include/utils/SymmetricRankTwoTensor.h:147]: I... can't remember. I'll just revert that line. This was during a time I was finding warnings with the new compilers. Since I've backed out of bumping the compilers, that work is no longer needed.

File: modules/contact/test/tests/mortar_cartesian_lms/tests (hunk 3 above)
Comment (milljm) [reply modules/contact/test/tests/mortar_cartesian_lms/tests:13]: Good idea about the "full `next` complement".

Comment (milljm) [reply modules/contact/test/tests/mortar_cartesian_lms/tests:13]: This test is fine.

I think what happened is that I mistook a faulty test for an actual MPI version issue (which I redacted several commits ago).

**Comment by loganharbour** [comment]
I hate to add on to this Jason, but can you add:

```
--download-hdf5-configure-arguments="--enable-cxx"
```

to the petsc configure here?

**Comment by milljm** [comment]
@YaqiWang @NamjaeChoi 

When I enable said option, HDF5 fails to configure:

```pre
configure:31599: checking for API tracing
configure:31627: result: no
configure:31642: checking whether a memory checking tool will be used
configure:31668: result: no
configure:31680: checking whether internal memory allocation sanity checking is used
configure:31706: result: no
configure:31816: error: --enable-cxx and --enable-parallel flags are incompatible. Use --enable-unsupported to override this error.
```

Perhaps this better belongs in a separate PR while we diagnose?

**Comment by NamjaeChoi** [comment]
> @YaqiWang @NamjaeChoi
> 
> When I enable said option, HDF5 fails to configure:
> 
> ```
> configure:31599: checking for API tracing
> configure:31627: result: no
> configure:31642: checking whether a memory checking tool will be used
> configure:31668: result: no
> configure:31680: checking whether internal memory allocation sanity checking is used
> configure:31706: result: no
> configure:31816: error: --enable-cxx and --enable-parallel flags are incompatible. Use --enable-unsupported to override this error.
> ```
> 
> Perhaps this better belongs in a separate PR while we diagnose?

This is not urgent, so no problem with having it as a separate PR. Thanks.

**Comment by milljm** [comment]
@NamjaeChoi @YaqiWang 

Because you both thumbs upped the request, I am assuming your are using said feature now. How were you able to get HDF5 built with said features? No matter what I try it appears I never end up with an HDF5 built with a CXX/Parallel combination.

**Comment by NamjaeChoi** [comment]
> @NamjaeChoi @YaqiWang
> 
> Because you both thumbs upped the request, I am assuming your are using said feature now. How were you able to get HDF5 built with said features? No matter what I try it appears I never end up with an HDF5 built with a CXX/Parallel combination.

I didn't install anything, I just included `H5Cpp.h` and it worked, so I thought it was already available. @loganharbour's thought is that the it got installed as a dependency for some other packages in the conda environment and I happened to use it.

File: conda/petsc/build.sh [hunk 10]
```diff
@@ -92,8 +92,10 @@ make
 # FIXME: Workaround mpiexec setting O_NONBLOCK in std{in|out|err}
 # See https://github.com/conda-forge/conda-smithy/pull/337
 # See https://github.com/pmodels/mpich/pull/2755
```
Comment (cticenhour) [inline conda/petsc/build.sh:94]:
```suggestion
```

In conjunction with the other change here.

File: conda/petsc/build.sh [hunk 11]
```diff
@@ -92,8 +92,10 @@ make
...
-make check MPIEXEC="${RECIPE_DIR}/mpiexec.sh"
-
+# Skip for non-linux platforms (testing halts on ARM for some unknown reason)
+if [ `uname` == 'linux' ]; then
+  make check MPIEXEC="${RECIPE_DIR}/mpiexec.sh"
+fi
```
Comment (cticenhour) [inline conda/petsc/build.sh:98]:
```suggestion
  make check
```
Let's try simply this out - it worked great for me in #23502

**Review by cticenhour — CHANGES_REQUESTED** [review]

**Comment by NamjaeChoi** [comment]
> > @NamjaeChoi @YaqiWang
> > Because you both thumbs upped the request, I am assuming your are using said feature now. How were you able to get HDF5 built with said features? No matter what I try it appears I never end up with an HDF5 built with a CXX/Parallel combination.
> 
> I didn't install anything, I just included `H5Cpp.h` and it worked (in my local), so I thought it was already available. @loganharbour's thought is that it got installed as a dependency for some other packages in the MOOSE conda environment and I happened to use it. I'm not sure if I'm actually using parallel HDF5.

File: conda/peacock/meta.yaml [hunk 12]
```diff
@@ -1,36 +1,40 @@
...
-{% set version = "2022.12.05" %}
+{% set version = "2023.08.17" %}
 
 package:
   name: moose-peacock
   version: {{ version }}
 
 source:
-  path: ../peacock
+  - path: ../peacock
```
[the remark this answers is not in the document]
Comment (milljm) [reply conda/peacock/meta.yaml:14]: Ah, nice catch. Earlier renditions of this PR included multiples (trame).

File: apptainer/mpich.def [hunk 13]
```diff
@@ -84,7 +84,11 @@ EOF
 
     # Create a pretty prompt (ensure last item sourced)
     cat <<'EOF' > ${SINGULARITY_ROOTFS}/.singularity.d/env/99-zzz_prompt.sh
-PS1="\[\033[1;34m\][`basename ${APPTAINER_NAME:-$SINGULARITY_NAME} .sif`]\[\033[1;32m\][\w]\[\033[0m\]> "
+if [ -n "${CUSTOM_PROMPT}" ]; then
+    PS1=${CUSTOM_PROMPT}
+else
+    PS1="\[\033[1;34m\][`basename ${APPTAINER_NAME:-$SINGULARITY_NAME} .sif`]\[\033[1;32m\][\w]\[\033[0m\]> "
+fi
```
Comment (cticenhour) [inline apptainer/mpich.def:91]: 🎉

**Review by cticenhour — APPROVED** [review]
